src/widgets/voice-search/voice-search.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { render, unmountComponentAtNode } from 'preact-compat';
+import cx from 'classnames';
+import {
+  getContainerNode,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { component } from '../../lib/suit';
+import connectVoiceSearch, {
+  VoiceSearchRenderer,
+} from '../../connectors/voice-search/connectVoiceSearch';
+import VoiceSearch from '../../components/VoiceSearch/VoiceSearch';
+import defaultTemplates from './defaultTemplates';
+import { Template } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'voice-search' });
+const suit = component('VoiceSearch');
+
+export type VoiceSearchCSSClasses = {
+  root: string;
+  button: string;
+  status: string;
+};
+
+export type VoiceSearchTemplates = {
+  buttonText: Template<object>;
+  status: Template<object>;
+};
+
+export type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+type VoiceSearchWidgetParams = {
+  container: string | HTMLElement;
+  cssClasses: VoiceSearchCSSClasses;
+  templates: VoiceSearchTemplates;
+  searchAsYouSpeak?: boolean;
+};
+
+type VoiceSearchWidgetRenderParams = {
```

Comment:
This type is not useful, those information are already available through the renderer. The type that we have to define here is the `widgetParams` that is provided to the `render` function. The type is different from the one provided to the widget because some values are now required (like `cssClasses` & `templates`). You can take example on the [`QueryRulesCustomData`](https://github.com/algolia/instantsearch.js/blob/577e4ccb33155d0d4678b04505132d39a2918f22/src/widgets/query-rule-custom-data/query-rule-custom-data.tsx#L22-L36) widget.

Reply:
Thanks for the explanation and the example link! 👍